ports/boost-modular-build-helper/CMakeLists.txt

```
@@ -120,57 +141,115 @@ if(NOT LDFLAGS STREQUAL "")
-        set(CXXFLAGS "${CXXFLAGS} <compileflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN}${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
-        set(LDFLAGS "${LDFLAGS} <linkflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN}${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
+        string(APPEND CXXFLAGS " <compileflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN}${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
+        string(APPEND LDFLAGS " <linkflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN}${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
     else()
-        set(CXXFLAGS "${CXXFLAGS} <compileflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN} <compileflags>${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
-        set(LDFLAGS "${LDFLAGS} <linkflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN} <linkflags>${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
+        string(APPEND CXXFLAGS " <compileflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN} <compileflags>${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
+        string(APPEND LDFLAGS " <linkflags>${CMAKE_CXX_COMPILE_OPTIONS_EXTERNAL_TOOLCHAIN} <linkflags>${CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN}")
     endif()
 endif()
+
 if(CMAKE_SYSROOT AND CMAKE_CXX_COMPILE_OPTIONS_SYSROOT)
-    set(CXXFLAGS "${CXXFLAGS} <compileflags>${CMAKE_CXX_COMPILE_OPTIONS_SYSROOT}${CMAKE_SYSROOT}")
-    set(LDFLAGS "${LDFLAGS} <linkflags>${CMAKE_CXX_COMPILE_OPTIONS_SYSROOT}${CMAKE_SYSROOT}")
+    string(APPEND CXXFLAGS " <compileflags>${CMAKE_CXX_COMPILE_OPTIONS_SYSROOT}${CMAKE_SYSROOT}")
+    string(APPEND LDFLAGS " <linkflags>${CMAKE_CXX_COMPILE_OPTIONS_SYSROOT}${CMAKE_SYSROOT}")
 endif()
-foreach(INCDIR ${CMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES})
-    set(CXXFLAGS "${CXXFLAGS} <compileflags>${CMAKE_INCLUDE_FLAG_C}${CMAKE_INCLUDE_FLAG_C_SEP}${INCDIR}")
+foreach(INCDIR IN LISTS CMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES)
+    string(APPEND CXXFLAGS " <compileflags>${CMAKE_INCLUDE_FLAG_C}${CMAKE_INCLUDE_FLAG_C_SEP}${INCDIR}")
 endforeach()
 
 if(APPLE)
-  set(CXXFLAGS "${CXXFLAGS} <compileflags>-D_DARWIN_C_SOURCE <cxxflags>-std=c++11 <cxxflags>-stdlib=libc++")
-  set(LDFLAGS "${LDFLAGS} <linkflags>-stdlib=libc++")
+  string(APPEND CXXFLAGS " <compileflags>-D_DARWIN_C_SOURCE <cxxflags>-std=c++11 <cxxflags>-stdlib=libc++")
+  string(APPEND LDFLAGS " <linkflags>-stdlib=libc++")
 endif()
 
-find_library(ZLIB_LIBPATH z)
-list(APPEND B2_OPTIONS
-    -sZLIB_BINARY=z
-    -sZLIB_LIBPATH="${ZLIB_LIBPATH}"
-)
-if(CMAKE_BUILD_TYPE STREQUAL "Release")
-    find_library(BZIP2_LIBPATH bz2)
-    list(APPEND B2_OPTIONS
-        -sBZIP2_BINARY=bz2
-        -sBZIP2_LIBPATH="${BZIP2_LIBPATH}"
-        variant=release
-    )
-elseif(CMAKE_BUILD_TYPE STREQUAL "Debug")
-    find_library(BZIP2_LIBPATH bz2d)
+set(TOOLSET_OPTIONS)
+if(CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+    # cl in b2 appears to not receive `LIBPATH` for finding winmd files, so we transform them to `/AI` options.
+    set(libpath_args "$ENV{LIBPATH}")
+    # Apply: {x -> /AI"x"}

```

```suggestion
    # Apply: x -> /AI"x"

```

I think the curly braces obfuscate rather than clarify. I may just be bad at reading though. Up to you :)

---

I added the curlies because the below replacements end in `\` and I didn't want to accidentally cause line termination escaping issues

---

You and Nicole are both killing me today.